### ISSUE TITLE
undefined function with js-interop

### ISSUE THREAD

**Issue by Kleak** [dart-lang/sdk#26286]
Problem:

in this [file](https://gitlab.com/grimshield/nodejs.dart/blob/master/lib/src/fs.dart) i have a methods name access in the class Fs and when i compile the code to js an error is throw telling me that there is an undefined function...

I have try to name it for example access1 and it's working but not really pretty to use.

**Comment by Kleak** [comment]
ok it work now but i don't know why ...

this if the method before : https://gitlab.com/grimshield/nodejs.dart/blob/b76b8719f0977b5640a998c46cc55ef227710c87/lib/src/fs.dart#L87

and this is the method now : 
https://gitlab.com/grimshield/nodejs.dart/blob/9e0a384f3b8d8d5e25b71bf491d9c9ad1710f3e4/lib/src/fs.dart#L87

**Comment by kevmoo** [comment]
Maybe an issue with compiling async code?

**Comment by sigmundch** [comment]
@kmillikin - does this raise any immediate ideas for you?

@Kleak - IIUC the diff, is it basically that if you have:
```dart
Future<FsAccessSyncError> access(String path, [int options = 0]) async =>
      await _access_internal(path, options);
```
you see the error, but if you remove the trivial async/await:
```dart
Future<FsAccessSyncError> access(String path, [int options = 0])  =>
       _access_internal(path, options);
```
then it works?

**Comment by Kleak** [comment]
@sigmundch yes exactly but there is a lot of way to write this and where i get the same error :/

```
Future<AccessSyncError> access(String path, [int options = 0]) {
    Completer completer = new Completer();
    _access_internal(path, options, (AccessSyncError error) {
      completer.complete(error);
    });
    return completer.future;
  }
```

this one raise an error when you use the function with await
It seems that it's when await is used with a Completer...

Dart SDK 1.15.0
